src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -143,3 +147,122 @@ def _check_project_area_from_url_params_fields_exist(
             raise Exception('project area field, "polygons" is an empty list')
         if 'id' not in project_area.keys():
             raise Exception('project area missing field, "id"')
+
+
+# Forsys input dataframe headers.
+class ForsysInputHeaders():
+    # Constant headers for project and stand ID's, area, and cost.
+    FORSYS_PROJECT_ID_HEADER = "proj_id"
+    FORSYS_STAND_ID_HEADER = "stand_id"
+    FORSYS_AREA_HEADER = "area"
+    FORSYS_COST_HEADER = "cost"
+
+    # Header prefixes for conditions and priorities.
+    _CONDITION_PREFIX = "c_"
+    _PRIORITY_PREFIX = "p_"
+
+    # List of headers for priorities.
+    # Downstream, this must be in the same order as constructor input
+    # priorities.
+    priority_headers: list[str]
+
+    def __init__(self, priorities: list[str]) -> None:
+        self.priority_headers = []
+
+        for p in priorities:
+            self.priority_headers.append(self.priority_header(p))
+
+    # Returns a priority header givn a priority string.
+    def priority_header(self, priority: str) -> str:
+        return self._PRIORITY_PREFIX + priority
+
+    # Reteurns a condition hader given a condition string.
+    def condition_header(self, condition: str) -> str:
+        return self._CONDITION_PREFIX + condition
+
+
+class ForsysProjectAreaRankingInput():
+    # dictionary representing a forsys input dataframe.
+    forsys_input: dict[str, list]
+
+    def __init__(
+            self, params: ForsysProjectAreaRankingRequestParams,
+            headers: ForsysInputHeaders) -> None:
+        region = params.region
+        priorities = params.priorities
+        project_areas = params.project_areas
+
+        condition_ids_to_names = self._get_condition_ids_to_names(
+            region, priorities)
+        conditions = self._get_conditions(condition_ids_to_names.keys())
+
+        self.forsys_input = self._initialize_forsys_input(headers, priorities)
+
+        for proj_id in project_areas.keys():
+            geo = self._get_raster_geo(project_areas[proj_id])
+
+            self.forsys_input[headers.FORSYS_PROJECT_ID_HEADER].append(proj_id)
+            self.forsys_input[headers.FORSYS_STAND_ID_HEADER].append(proj_id)
+            self.forsys_input[headers.FORSYS_AREA_HEADER].append(geo.area)
+            # TODO: figure out the right value for the units: 5000 is just a
+            # placeholder.
+            self.forsys_input[headers.FORSYS_COST_HEADER].append(
+                geo.area * 5000)
+            for c in conditions:
+                name = condition_ids_to_names[c.condition_dataset_id]
+                score = compute_condition_score_from_raster(
+                    geo, c.raster_name)
+                if score is None:
+                    raise Exception(
+                        "no score was retrieved for condition, %s" % name)
+                self.forsys_input[headers.condition_header(name)].append(score)
+                # TODO: fix this to be sum(1.0 - score) rather than
```

Comment:
it looks like right now, this doesn't use sum() or mean(). should we be using that?

Reply:
This isn't right because the impact of a project area should be larger for larger areas. It should be a sum rather than an average.

Reply:
I can fix this (and also estimate area without worrying about units) by changing the postgis call - but that'll need to be in a subsequent cl.